src/main/java/io/strimzi/kafka/bridge/amqp/converter/AmqpDefaultMessageConverter.java

```diff
@@ -124,7 +125,7 @@ public Message toMessage(String address, ConsumerRecord<String, byte[]> record)
 		MessageAnnotations messageAnnotations = new MessageAnnotations(map);
 		message.setMessageAnnotations(messageAnnotations);
 		
-		message.setBody(new Data(new Binary(record.value())));
+		message.setBody(new Data(new Binary(record.record().value())));
```

Comment:
I think that calling `record()` is not necessary. The `value()` on the `KafkaConsumerRecord` return exactly the inner record value. Following its implementation in the Vert.x Kafka client library : 

```
public V value() {
    return this.record.value();
  }
```

Reply:
when testing AmqpSinkBridgeEndpointMockTest.normalFlow_AtLeastOnce gives a NullPointerException. In debugging record.value() is null. Here is the stack trace:

java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at io.strimzi.kafka.bridge.amqp.AmqpSinkBridgeEndpointMockTest.normalFlow_AtLeastOnce(AmqpSinkBridgeEndpointMockTest.java:238)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.lang.NullPointerException
	at org.apache.qpid.proton.amqp.Binary.<init>(Binary.java:36)
	at io.strimzi.kafka.bridge.amqp.converter.AmqpDefaultMessageConverter.toMessage(AmqpDefaultMessageConverter.java:129)
	at io.strimzi.kafka.bridge.amqp.converter.AmqpDefaultMessageConverter.toMessage(AmqpDefaultMessageConverter.java:41)
	at io.strimzi.kafka.bridge.amqp.AmqpSinkBridgeEndpoint.sendAmqpMessage(AmqpSinkBridgeEndpoint.java:222)
	at io.strimzi.kafka.bridge.SinkBridgeEndpoint.handleReceived(SinkBridgeEndpoint.java:515)
	at io.strimzi.kafka.bridge.SinkBridgeEndpoint.handleKafkaRecord(SinkBridgeEndpoint.java:321)

Reply:
@pushpithaDilhan this is because of  ```when(mockVertxRecord.record()).thenReturn(mockKafkaRecord);
``` in method ```mockRecord``` of class ```MockRecordProducer```. 
You can pass a KafkaConsumerRecord object directly in ```handler.invoke``` to prevent it. like this ```handler.invoke(endpoint, new KafkaConsumerRecordImpl(
				new ConsumerRecord("mytopic", 0, 0, null, "hello, world".getBytes())));```
As @ppatierno said, KafkaConsumerRecord.value returns record.value itself.

Reply:
@shubhamvashisht Thanks, actual problem was that method was using `ConsumerRecord` from both libraries. since `mockRecord` uses in multiple tests, it should be fixed inside the method instead of directly using the object.